File: backend/app/schemas/market.py
"""Pydantic schemas for market overview data."""

from pydantic import BaseModel


class IndexData(BaseModel):
    symbol: str
    name: str
    price: float
    change: float
    change_percent: float
    sparkline: list[float]


class SectorNewsItem(BaseModel):
    title: str
    url: str | None = None
    publisher: str | None = None


class SectorData(BaseModel):
    name: str
    etf: str
    price: float
    change_percent: float
    news: list[SectorNewsItem]


class MoverData(BaseModel):
    symbol: str
    name: str
    price: float
    change_percent: float


class MarketOverviewResponse(BaseModel):
    indices: list[IndexData]
    sectors: list[SectorData]
    gainers: list[MoverData]
    losers: list[MoverData]
    updated_at: str

File: frontend/types/market.ts
export interface IndexData {
  symbol: string
  name: string
  price: number
  change: number
  change_percent: number
  sparkline: number[]
}

export interface SectorNewsItem {
  title: string
  url: string | null
  publisher: string | null
}

export interface SectorData {
  name: string
  etf: string
  price: number
  change_percent: number
  news: SectorNewsItem[]
}

export interface MoverData {
  symbol: string
  name: string
  price: number
  change_percent: number
}

export interface MarketOverviewResponse {
  indices: IndexData[]
  sectors: SectorData[]
  gainers: MoverData[]
  losers: MoverData[]
  updated_at: string
}
